### PR TITLE
Cleaning up a couple bugs, adding configuration for search heads, and instance type updates

### PR DESCRIPTION
**Pull request by billbartlett**
*Issue #, if available:*

*Description of changes:*

- init.d script fixes.
- instance type updates and updated defaults
- fixed bug with SHC and all search heads conflicting with the same hostname
- larger minimum EBS volume requirements due to AMI construction, (both indexers and search heads)
- search head disk parameter and label added
- indexer cluster search factor parameter added, and limited from 2-4
- changed default replication factor to 2, and updated range to 2-4 from 2-5
- changed instance type of deployer and cluster master to c5 family

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: templates/splunk-enterprise.template [hunk 1]
```diff
@@ -91,6 +96,16 @@
             "Type": "AWS::EC2::Subnet::Id",
             "Default": ""
         },
+        "QSS3BucketName": {
```
Comment (sshvans) [inline templates/splunk-enterprise.template]:
```suggestion
        "QSS3BucketName": {
            "AllowedPattern": "^[0-9a-zA-Z]+([0-9a-zA-Z-]*[0-9a-zA-Z])*$",
            "ConstraintDescription": "Quick Start bucket name can include numbers, lowercase letters, uppercase letters, and hyphens (-). It cannot start or end with a hyphen (-).",
            "Default": "aws-quickstart",
            "Description": "S3 bucket name for the Quick Start assets. Quick Start bucket name can include numbers, lowercase letters, uppercase letters, and hyphens (-). It cannot start or end with a hyphen (-).",
            "Type": "String"
        },
```

File: templates/splunk-enterprise.template [hunk 2]
```diff
@@ -91,6 +96,16 @@
...
+            "Default": "splk-quickstart-testing",
+            "Description": "S3 bucket name for the Quick Start assets.",
+            "Type": "String"
+        },
+        "QSS3KeyPrefix": {
```
Comment (sshvans) [inline templates/splunk-enterprise.template]:
```suggestion
       "QSS3KeyPrefix": {
            "AllowedPattern": "^([0-9a-zA-Z-.]+/)*$",
            "ConstraintDescription": "Quick Start key prefix can include numbers, lowercase letters, uppercase letters, hyphens (-), dots (.) and forward slash (/). The prefix should end with a forward slash (/).",
            "Default": "quickstart-splunk-enterprise/",
            "Description": "S3 key prefix for the Quick Start assets. Quick Start key prefix can include numbers, lowercase letters, uppercase letters, hyphens (-), dots (.) and forward slash (/) and it should end with a forward slash (/).",
            "Type": "String"
        },
```

Comment (billbartlett) [reply templates/splunk-enterprise.template]: looks good to me

**Comment by billbartlett** [comment]
Is everything good to go with these edits?  Wanted to make sure I didn't miss anything.

**Pull request merged**